src/imagery/i.eodag/testsuite/test_eodag.py

```diff
@@ -0,0 +1,435 @@
+############################################################################
+#
+# MODULE:      i.eodag
+#
+# AUTHOR(S):   Hamed Elgizery <hamedashraf2004 gmail.com>
+# MENTOR(S):   Luca Delucchi, Veronica Andreo, Stefan Blumentrath
+#
+# PURPOSE:     Tests i.eodag input parsing / searching results.
+#              Uses NC Full data set.
+#
+# COPYRIGHT:   (C) 2024-2025 by Hamed Elgizery, and the GRASS development team
+#
+#              This program is free software under the GNU General Public
+#              License (>=v2). Read the file COPYING that comes with GRASS
+#              for details.
+#
+#############################################################################
+
+import re
+import json
+from datetime import datetime
+from grass.gunittest.case import TestCase
+from grass.gunittest.gutils import get_current_mapset, is_map_in_mapset
+from grass.exceptions import CalledModuleError
+from grass.pygrass.modules import Module
+from subprocess import PIPE
+
+
+class TestEodag(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """Use temporary region settings."""
+        cls.use_temp_region()
+        cls.runModule(
+            "v.extract",
+            input="urbanarea",
+            where="NAME = 'Durham'",
+            output="durham",
+            overwrite=True,
+        )
+        cls.peps_up = True
+        cls.cop_dataspace_up = True
+        cls.creodias_up = True
+        cls.planetary_computer_up = True
+        try:
+            cls.runModule(
+                "i.eodag",
+                flags="l",
+                provider="peps",
+                producttype="S2_MSI_L1C",
+                start="2024-01-01",
+                end="2024-01-01",
+                map="durham",
+                quiet=True,
+            )
+        except:
+            print(
+                "Faild to connect to provider 'peps', all testes associated with 'peps' will pass unconditionally."
+            )
+            cls.peps_up = False
```

Comment:
This is a very good approach!

Please consider:
a) using EODAGs Python API with a minimal search, to test if the provider is available. That way we do not have to assume the `i.eodag -l` code is  correct.
b) you could test provider availability in a loop to reduce code duplication
c) With_cls.peps_up_ you could use the skipIf-method from unittest: https://docs.python.org/3/library/unittest.html#skipping-tests-and-expected-failures To skip tests that will fail because a provider is not available (or temporary down)...